games-fps/chocolate-doom/chocolate_doom-3.0.0.recipe

```diff
@@ -0,0 +1,76 @@
+SUMMARY="Doom source port that accurately reproduces the experience of Doom"
+DESCRIPTION="Chocolate Doom is a source port of the game Doom, by id Software. \
+The purpose of Chocolate Doom is to be as compatible as possible with the \
+original DOS version of Doom."
+HOMEPAGE="https://www.chocolate-doom.org/"
+COPYRIGHT="Id Software Inc. 1993-1996
+	Simon Howard
+	James Haley
+	Samuel Villarreal"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://www.chocolate-doom.org/downloads/$portVersion/chocolate-doom-$portVersion.tar.gz"
+CHECKSUM_SHA256="73aea623930c7d18a7a778eea391e1ddfbe90ad1ac40a91b380afca4b0e1dab8"
+SOURCE_DIR="chocolate-doom-$portVersion"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	chocolate_doom$secondaryArchSuffix = $portVersion
+	cmd:chocolate_doom$secondaryArchSuffix = $portVersion
```

Comment:
I believe some of these are GUI applications. If that's the case, you would want to create a resources file and install the binary to `$appsDir`

Reply:
Should I create a resources file for each executable, seeing that they're all for separate games?

Reply:
You could make a file containing placeholders (ie. `@MAJOR@`, `@DESC@`, `@ICON@`) then fill them in during packaging with `sed`.

Reply:
Also, if they're separate games, you should consider splitting the package into smaller ones, each containing a different version of the game (assuming they don't depend on each other).

Reply:
I did find other recipe and resource files that did what you said with the placeholders and implemented it in my recipe.

I had a few issues with putting the files into $appDir, however.  First it removed the apps from the global path (which I'm guessing could be fixed with some script to add $appsDir to Path), but also you need to specify a wad file as an argument, which made it not launch when clicking from a deskbar symlink.  

Also, I added the defineDebugInfoPackage info the way the page you linked me told me to, but when I did it also gave me an error once it was done compiling saying it couldn't find the '/chocolate[-,_]doom' file.  If I removed that section, it compiled and installed. 

Also, I noticed the source itself includes a png to use as an icon, but even with manually copying it beforehand to the additional-files dir, adding it to my recipe file, and adding code to add the png to the executable, nothing happened.  I searched all over, but could not find a way to convert the png to a byte string used in the resource file's 'resource vector_icon'.  Icon-O-Matic would not open the file correctly.    

Thank you for the nudges in the right direction.  I've learned more than I could have ever learned to with just your suggestions!

Reply:
Oh, and from what I saw of the output from the compile, it does not appear they are separate games, but more so each executable runs the iwad files from their respective games (doom, heretic, hexen, strife).  I can't tell you exactly, but each game might be a copy from the main executable.  The difference is in the respective setup files which have options specific to the game it's configuring.   

After creating the resource file for the main chocolate-doom executable, it did not add the resource info to the other executables (as would be expected), but I feel they should all have their own resource files)

Reply:
I think I know why haikuporter dropped the resources when you had the `defineDebugInfoPackage`. It's (probably) because you are using the latest stable release of haikuporter.

BTW, in case you would like to use `defineDebugInfoPackage` with a package which creates runtimes that have resources, you should use a git revision of haikuporter that includes [this commit](https://github.com/haikuports/haikuporter/commit/c4c7fe9307a7f35ac05a02aa34bcb9b527c03e45).

(Maybe we should tag the current HEAD of HaikuPorter as release "1.2.3".)